### PR TITLE
Feature/clickhouse connector

### PR DESCRIPTION
**Pull request by MahmoudElhalwany**
#1085

### CODE REVIEW

File: chaos_genius/connectors/clickhouse.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        port = int(db_info.get("port"))
+        username = db_info.get("username")
+        database = db_info.get("database")
+        password = db_info.get("password")
+        if not (host and port and username and password and database):
+            raise NotImplementedError(
+                "Database Credential not found for ClickHouse."
+            )
+        self.sqlalchemy_db_uri = (
+            f"clickhouse+http://{username}:{password}@{host}:{port}/{database}"
```
Comment (Samyak2) [inline chaos_genius/connectors/clickhouse.py:35]: Is there a specific reason we have used `http` over the `native` driver which this library recommends using: https://github.com/xzkostyan/clickhouse-sqlalchemy#usage?

File: chaos_genius/utils/metadata_api_config.py [hunk 2]
```diff
@@ -66,6 +67,13 @@
         "supported_aggregations": ["mean", "sum", "count"],
         "supports_multidim_dd": True
     },
+    "ClickHouse": {
+        "tables": True,
+        "views": False,
+        "materialized_views": False,
```
Comment (Samyak2) [inline chaos_genius/utils/metadata_api_config.py:73]: Looks like views and materialized views are both supported by clickhouse: https://clickhouse.com/docs/en/sql-reference/statements/create/view/

We can set these to `True`

File: requirements/prod.txt [hunk 3]
```diff
@@ -23,6 +23,7 @@ sqlalchemy-redshift==0.8.6
 pydruid[sqlalchemy]~=0.6.2
 PyAthena[SQLAlchemy,Pandas]==2.5.2
 sqlalchemy-databricks==0.2.0
+clickhouse-sqlalchemy==0.2.1
```
Comment (Samyak2) [inline requirements/prod.txt:26]: A new version was published just yesterday :smile:

Let's update this to v0.2.2. - https://github.com/xzkostyan/clickhouse-sqlalchemy/releases/tag/0.2.2

**Review by Samyak2 — CHANGES_REQUESTED** [review]
Hey @MahmoudElhalwany, thank you very much for your contribution! This looks great overall and everything seems to be working (I tested on a local clickhouse set up). The icon is good too.

That said, I had a few comments and questions on the changes. Please find them below.

File: chaos_genius/connectors/clickhouse.py (hunk 1 above)
Comment (MahmoudElhalwany) [reply chaos_genius/connectors/clickhouse.py:35]: Hello @Samyak2 , When i used the native client there were some errors during the mapping of dataTypes, but we can try again with the native client and see what in its bag ?

File: chaos_genius/utils/metadata_api_config.py (hunk 2 above)
Comment (MahmoudElhalwany) [reply chaos_genius/utils/metadata_api_config.py:73]: Sure, here we can set them to True

File: requirements/prod.txt (hunk 3 above)
Comment (MahmoudElhalwany) [reply requirements/prod.txt:26]: hahaha, okay we can update it to 0.2.2

Comment (MahmoudElhalwany) [reply requirements/prod.txt]: now u can check them -- thanks

**Review by Samyak2 — COMMENTED** [review]
Looks great!

It would be helpful if you could show the error you got when trying the native client. I'll try it from my side too.

**Comment by MahmoudElhalwany** [comment]
> Looks great!
> 
> It would be helpful if you could show the error you got when trying the native client. I'll try it from my side too.

Okay, but there is a potential bug too `there is some datatypes in ClickHouse not supported in the library and in python like Int128 `

**Comment by Samyak2** [comment]
> there is some datatypes in ClickHouse not supported in the library and in python like Int128

Looks like there is an open PR for this: https://github.com/xzkostyan/clickhouse-sqlalchemy/pull/184

Is this something that is necessary for your usecase?

---

Did you find any bugs specific to the native interface/connector?

**Comment by MahmoudElhalwany** [comment]
> > there is some datatypes in ClickHouse not supported in the library and in python like Int128
> 
> Looks like there is an open PR for this: [xzkostyan/clickhouse-sqlalchemy#184](https://github.com/xzkostyan/clickhouse-sqlalchemy/pull/184)
> 
> Is this something that is necessary for your usecase?
> 
> Did you find any bugs specific to the native interface/connector?

@Samyak2 
I just have some tables that use Int128 in their columns, it gives me errors when I choose that database when initialization.
for native bugs -- until now, No. So, we can change it to native.
Did you find any bugs using the native ??

**Review by Samyak2 — APPROVED** [review]
Looks good! Thank you for contributing this connector. I'm sure the community will benefit greatly from this.

We'll update `clickhouse-sqlalchemy` to the next version which will likely include the fixes for `Int128` when it's released.

**Comment by MahmoudElhalwany** [comment]
> hich will likely include the fixes for `Int128` when it's relea

thanks -- this is my first real contribution to open source project -- so I must make it an anniversary 😄

**Pull request merged**